contracts/prize-strategy/PrizeSplit.sol

```diff
@@ -0,0 +1,195 @@
+    return _prizeSplits;
+  }
+
+  /**
+    * @notice Read a prize split config by index.
+    * @dev Read a PrizeSplitConfig via an active _prizeSplits index.
+    * @param prizeSplitIndex The target index to read
+    * @return PrizeSplitConfig A single prize split config
+  */
+  function prizeSplit(uint256 prizeSplitIndex) external view returns (PrizeSplitConfig memory) {
+    return _prizeSplits[prizeSplitIndex];
+  }
+
+  /**
+    * @notice Update, add and/or remove prize split(s) configuration.
+    * @dev Update, add and/or remove prize split configs via an array of PrizeSplitConfig structs. Removes PrizeSplitConfig(s) if array lengths don't match. Limited to contract owner.
+    * @param newPrizeSplits Array of PrizeSplitConfig structs
+  */
+  function setPrizeSplits(PrizeSplitConfig[] calldata newPrizeSplits) external onlyOwner {
+    uint256 newPrizeSplitsLength = newPrizeSplits.length;
+
+    // Add and/or update prize split configs using newPrizeSplits PrizeSplitConfig structs array.
+    for (uint256 index = 0; index < newPrizeSplitsLength; index++) {
+      PrizeSplitConfig memory split = newPrizeSplits[index];
+      require(split.token <= 1, "MultipleWinners/invalid-prizesplit-token");
+      require(split.target != address(0), "MultipleWinners/invalid-prizesplit-target");
+      
+      if (_prizeSplits.length <= index) {
+        _prizeSplits.push(split);
+      } else {
+        PrizeSplitConfig memory currentSplit = _prizeSplits[index];
+        if (split.target != currentSplit.target || split.percentage != currentSplit.percentage || split.token != currentSplit.token) {
+          _prizeSplits[index] = split;
+        } else {
+          continue;
+        }
+      }
+
+      // Emit the added/updated prize split config.
+      emit PrizeSplitSet(split.target, split.percentage, split.token, index);
+    }
+
+    // Remove prize splits by comparing the current _prizesSplits with the passed prizeSplits
+    while (_prizeSplits.length > newPrizeSplitsLength) {
+      uint256 _index = _prizeSplits.length.sub(1);
+      _prizeSplits.pop();
+      emit PrizeSplitRemoved(_index);
+    }
+
+    // Require the current prize split percentage does not exceed 100%.
+    uint256 totalPercentage = _totalPrizeSplitPercentageAmount();
+    require(totalPercentage <= 1000, "MultipleWinners/invalid-prizesplit-percentage-total");
+  }
+
+  /**
+    * @notice Updates a previously set prize split config.
+    * @dev Updates a prize split config by passing a new PrizeSplitConfig struct and current index position. Limited to contract owner.
+    * @param prizeStrategySplit PrizeSplitConfig config struct
+    * @param prizeSplitIndex The target index to update
+  */
+  function setPrizeSplit(PrizeSplitConfig memory prizeStrategySplit, uint8 prizeSplitIndex) external onlyOwner {
+    require(prizeSplitIndex <= _prizeSplits.length.sub(1), "MultipleWinners/nonexistent-prizesplit");
```

Comment:
This is funny- just go prizeSplitIndex < _prizeSplits.length.  Saves a touch of gas